bindings/node/lib/clientEncryption.js

```diff
@@ -42,7 +42,34 @@ module.exports = function(modules) {
      */
     createDataKey(provider, options, callback) {
       if (typeof options === 'function') (callback = options), (options = {});
-      options = options || {};
+      options = Object.assign({}, options);
+
+      // To avoid using libbson inside the bindings, we pre-serialize
+      // any keyAltNames here.
+      if (options.keyAltNames) {
+        if (!Array.isArray(options.keyAltNames)) {
+          throw new TypeError(
+            `Option "keyAltNames" must be an array of string, but was of type ${typeof options.keyAltNames}.`
+          );
+        }
+        const bson = this._bson;
+        const serializedKeyAltNames = [];
+        for (let i = 0; i < options.keyAltNames.length; i += 1) {
+          const item = options.keyAltNames[i];
+          const itemType = typeof item;
+          if (itemType !== 'string') {
+            throw new TypeError(
+              `Option "keyAltNames" must be an array of string, but item at index ${i} was of type ${itemType} `
+            );
+          }
+
+          serializedKeyAltNames.push(bson.serialize({ keyAltName: item }));
+        }
+
+        options.keyAltNames = serializedKeyAltNames;
+      } else {
+        delete options.keyAltNames;
```

Comment:
I think we shouldn't delete anything from a user passed in object, let's just ignore it if its not a type we care about.

Reply:
The specific reason I am deleting this is b/c of cases where the property exists, but the value is falsey. Specifically, the following two cases:

```js
// undefined value
{ keyAltNames: undefined }

// null value
{ keyAltNames: null }
```

In both cases, the bindings `Nan::Has` call will return true, b/c the object *has* the property, even though the value is falsey. This would later trigger a hard-to-trace error inside of the binding, and might be confusing to users. IMO it makes sense to delete the property off of the copy so we do not think it is there when checking inside the C++ code.

To make it a little more clear, maybe I could change the check to:

```js
else if (options.keyAltNames == null)
```

so it is clearly for the case of `null` and `undefined`?